app/scripts/controllers/service.js

```diff
@@ -27,11 +29,48 @@ angular.module('openshiftConsole')
       }
     ];
 
+    $scope.podFailureReasons = {
+      "Pending": "This pod will not receive traffic until all of its containers have been created."
+    };
+
     var watches = [];
 
+    // receives routes for the current service and maps a service port to each route name
+    var getPortsForRoutes = function(service, routes) {
+      if (!service || !routes) {
+        return;
+      }
+
+      $scope.portsForRoutes = {};
+      var orderedPorts = $filter("orderBy")(service.spec.ports, "port");
+
+      for(var i = 0; i < routes.length; i++) {
+        $scope.portsForRoutes[routes[i].metadata.name] = orderedPorts[i];
```

Comment:
This block doesn't seem right to me.  A route can specify what port it uses.  You might have multiple routes targeting the same ports on the service.  You might have some service ports that have no route associated with it.